src/App.tsx

```diff
@@ -61,14 +63,19 @@ export const App: React.FC = () => {
       filteredMarkets.map(currentMarket => {
         return getData(`mktDetail?id=${currentMarket.id}`)
           .then(response => checkForError(response))
-          .then(response => response.json());
+          .then(response => response.json())
+          .then(data => cleanDetailsData(data.marketdetails, currentMarket.id))
       })
     )
-      .then(arrayOfPromises => cleanDetailsData(arrayOfPromises))
-      .then(cleanData => setDetails(cleanData))
-      .catch(error => setError(error));
+    .then(data => addScheduleToMarkets(data, filteredMarkets))
+    .then(completeData => setData(completeData))
   };
 
+  const setData = (data: any) => {
```

Comment:
Is data type 'any' the best type for this?

Reply:
NOPE! Good catch! Corrected now!